Omr installation docs (#39593)
openshift/openshift-docs: installing/installing-mirroring-installation-images.adoc

 modules/installation-performing-disconnected-mirror-without-registry.adoc[leveloffset=+2]
////

[id="mirror-registry"]
== Mirror registry for {product-title}

This section introduces how to install mirror registry for {product-title}, a tool that holds mirrors of {product-title} images.

.Prerequisites

* An {product-title} subscription.
* {op-system-base-full} 8 or Fedora CoreOS with Podman 3.3 installed.
* About 12 GB for {product-title} 4.10 Release images, or about 358 GB for {product-title} 4.10 Release images and {product-title} 4.10 Operator images.

> * Fully qualified domain name for the Red Hat Quay service, which must resolve with a DNS server or your `/etc/hosts` file.

Author: I can't use via in OCP docs, in adherence with IBMSG. I can use 'through' or 'by' or something similar. Currently using 'through'

* Passwordless `sudo` access on the target host.

> * Key-based SSH connectivity on the target host, which are set up automatically for local installs. For remote hosts, you must generate your own SSH keys.

Author: Slight alteration, as OCP docs highly recommend avoiding parentheticals.

 modules/installation-performing-disconnected-mirror-without-registry.adoc[leveloffset=+2]
////

[id="mirror-registry"]
== Mirror registry for {product-title}

This section introduces how to install mirror registry for {product-title}, a small and streamlined container registry to use as a target for mirroring the required container images of {product-title} for disconnected installations.

.Prerequisites

* An {product-title} subscription.

> * {op-system-base-full} 8 or Fedora CoreOS with Podman 3.3 installed.

Reviewer: Should we list Fedora CoreOS? We don't support this OS, in our standard support channels, so is this something we want to 'support' or 'invite' support on. 

cc: @dmesser

> * About 12 GB for {product-title} 4.10 Release images, or about 358 GB for {product-title} 4.10 Release images and {product-title} 4.10 Red Hat Operator images.

Reviewer: Can we explain why this recommendation is the way it is? 
Is the 12 GB for the Release Payload and Related Images; for 1 version (x.y.z) or does this cover N (releases, z streams)? 

I think we need to elaborate on how we 'arrived' at these numbers, as these statement (as is) just invite questions (or a support burden).

> * Fully qualified domain name for the Red Hat Quay service, which must resolve through a DNS server or your `/etc/hosts` file.

Reviewer: `/etc/hosts` on the system running Quay, or an 'external' system? 
I think its simpler to just require an FQDN in / from a proper dns system, I recommend us striking the `/etc/hosts` recommendation.

 modules/installation-performing-disconnected-mirror-without-registry.adoc[leveloffset=+2]
////

[id="mirror-registry"]
== Mirror registry for {product-title}

This section introduces how to install mirror registry for {product-title}, a tool that holds mirrors of {product-title} images.

Mirror registry allows users to install a small-scale version of Red Hat Quay and its required components using the `mirror-registry` command line interface (CLI) tool. Mirror registry comes equipped with pre-configured local storage and a local database. It also includes pre-configured user credentials and access permissions with a single set of inputs and no additional configuration choices to get started.

Mirror registry provides a pre-determined network configuration and reports deployed component credentials and URIs upon success. A limited set of optional configuration inputs like fully qualified domain name (FQDN) services, superuser name and password, and custom TLS certificates are also provided. These services allow users to easily create an offline mirror of all {product-title} release content when running {product-title} in restricted network environments.

> Mirror registry is limited to hosting images that are required to install a disconnected {product-title} cluster. It uses local storage on your Red Hat Enterprise Linux (RHEL) machine, and storage supported by RHEL is supported by mirror registry. Mirror registry is not supported for production load, and is intended for use with {product-title} images only.

Reviewer 2: Can the references to RHEL be replaced with its attributes?

s/Red Hat Enterprise Linux (RHEL)/{op-system-base-full}
s/RHEL/{op-system-base}

Downloading mirror registry includes the following images:

* Quay: registry.redhat.io/quay/quay-rhel8:v3.6.1
* Redis: registry.redhat.io/rhel8/redis-6:1-25
* Postgres: registry.redhat.io/rhel8/postgresql-10:1-161

> * Mirror registry: quay.io/quay/mirror-registry-ee:latest

Reviewer 2: Lines 77-89 is a pretty large chunk of info that could probably be put into a conceptual module.

.Prerequisites

* An {product-title} subscription.

> * {op-system-first} 8 or Fedora with Podman 3.3 installed.

Reviewer 3: Not very sure about the value of `{op-system-first}`, but in [preview page](https://deploy-preview-39593--osdocs.netlify.app/openshift-enterprise/latest/installing/installing-mirroring-installation-images#mirror-registry), it gose to `Red Hat Enterprise Linux CoreOS (RHCOS) 8 or Fedora with Podman 3.3 installed.`.  For OMR, it should be `RHEL 8 or Fedora machine with podman v3.3 installed`

* About 12 GB for {product-title} 4.10 Release images, or about 358 GB for {product-title} 4.10 Release images and {product-title} 4.10 Operator images.
* Fully qualified domain name for the Red Hat Quay service, which must resolve with a DNS server or your `/etc/hosts` file.
* Passwordless `sudo` access on the target host.

> * Key-based SSH connectivity on the target host, which will be set up automatically for local installs. For remote hosts, you will need to generate your own SSH keys.

Reviewer 2: ```suggestion
* Key-based SSH connectivity on the target host, which will be set up automatically for local installs. For remote hosts, you must generate your own SSH keys.
```

* xref:../post_installation_configuration/preparing-for-users.adoc#post-install-mirrored-catalogs[Populating OperatorHub from mirrored Operator catalogs]

> 

Reviewer 2: These newlines should be removed
